### PR TITLE
Added a `--branch` parameter to Dolt global arguments on the command line.

### PR DESCRIPTION
**Pull request by Nithinps021**
Addresses https://github.com/dolthub/dolt/issues/6273

Thank you, @timsehn and @macneale4, for reviewing my [previous PR.](https://github.com/dolthub/dolt/pull/6507) I'm creating a new PR to incorporate the changes suggested by @macneale4.
 
Please let me know if anything more need to be added!

### CODE REVIEW

File: go/cmd/dolt/dolt.go [hunk 1]
```diff
@@ -688,7 +692,7 @@ var doc = cli.CommandDocumentationContent{
 	LongDesc:  `Dolt comprises of multiple subcommands that allow users to import, export, update, and manipulate data with SQL.`,
 
 	Synopsis: []string{
-		"<--data-dir=<path>> subcommand <subcommand arguments>",
+		"<--data-dir=<path>> <--branch=<branch name>>  subcommand <subcommand arguments>",
```
Comment (macneale4) [inline go/cmd/dolt/dolt.go:695]: Remove. I'll update the docs separately. This is out of date as it is.

File: go/cmd/dolt/dolt.go [hunk 2]
```diff
@@ -609,6 +609,12 @@ func buildLateBinder(ctx context.Context, cwdFS filesys.Filesys, mrEnv *env.Mult
 	var targetEnv *env.DoltEnv = nil
 
 	useDb, hasUseDb := apr.GetValue(commands.UseDbFlag)
+	useBranch, hasBranch := apr.GetValue(cli.BranchParam)
+
+	if hasUseDb && hasBranch{
+		dbName, _ := dsess.SplitRevisionDbName(useDb)
```
Comment (macneale4) [inline go/cmd/dolt/dolt.go:615]: When a user provide a branch in the DB and with the flag, they will get confusing results. You should error out when there is a branch in the db name that will be ignored.

**Review by macneale4 — CHANGES_REQUESTED** [review]
Thanks for this! A couple of minor comments to address. And new tests are needed in this file: https://github.com/dolthub/dolt/blob/main/integration-tests/bats/global-args.bats. Bats tests are pretty easy to run, but if you are unable to, they are execute in our PR approval checks.

**Comment by macneale4** [comment]
Tests for your change: https://github.com/dolthub/dolt/commit/65dcd39c3ed424cb814a9e2b06ba2f07c9583024

File: go/cmd/dolt/dolt.go (hunk 2 above)
Comment (Nithinps021) [reply go/cmd/dolt/dolt.go:615]: made the changes

**Comment by Nithinps021** [comment]
> Thanks for this! A couple of minor comments to address. And new tests are needed in this file: https://github.com/dolthub/dolt/blob/main/integration-tests/bats/global-args.bats. Bats tests are pretty easy to run, but if you are unable to, they are execute in our PR approval checks.

How to run these bats test locally?

File: go/cmd/dolt/dolt.go [hunk 3]
```diff
@@ -612,7 +612,10 @@ func buildLateBinder(ctx context.Context, cwdFS filesys.Filesys, mrEnv *env.Mult
 	useBranch, hasBranch := apr.GetValue(cli.BranchParam)
 
 	if hasUseDb && hasBranch{
-		dbName, _ := dsess.SplitRevisionDbName(useDb)
+		dbName, branchNameInDb := dsess.SplitRevisionDbName(useDb)
+		if len(branchNameInDb)!=0 {
+			cli.PrintErrf("The branch name in %s will be ignored.\n",useDb)
```
Comment (macneale4) [inline go/cmd/dolt/dolt.go:617]: We should fail in this case. Return the error so the user needs to provide reasonable arguments. Warning messages get ignored/missed.

**Review by macneale4 — CHANGES_REQUESTED** [review]

File: go/cmd/dolt/dolt.go (hunk 3 above)
Comment (Nithinps021) [reply go/cmd/dolt/dolt.go]: I have change it to return error, but still when i run bats test its failing

File: go/cmd/dolt/dolt.go [hunk 4]
```diff
@@ -614,7 +614,7 @@ func buildLateBinder(ctx context.Context, cwdFS filesys.Filesys, mrEnv *env.Mult
 	if hasUseDb && hasBranch{
 		dbName, branchNameInDb := dsess.SplitRevisionDbName(useDb)
 		if len(branchNameInDb)!=0 {
-			return nil, fmt.Errorf("The branch name in %s will be ignored. Remove the branch name from the DB name\n",useDb)
+			return nil, fmt.Errorf("Ambiguous branch name: %s or %s\n",branchNameInDb,useBranch)
```
Comment (macneale4) [inline go/cmd/dolt/dolt.go:617]: Take out the '\n'

**Comment by macneale4** [comment]
btw, I'll need to pull this into a branch I own so that the formatter test can pass.

**Comment by macneale4** [comment]
Pull this commit into your PR and I'll approve: https://github.com/dolthub/dolt/pull/6564/commits/e50cf43e776f7133da917717f4f23eaf7eb56c20

**Comment by Nithinps021** [comment]
@macneale4  Thanks for approving my PR. Is there any new issue that i can work on?

**Comment by macneale4** [comment]
So close! Pull this change in to your branch to address the merge conflict: https://github.com/dolthub/dolt/commit/4b8e0ee61436571f8c556cb1331f9df119ab14f8

Then I'll complete your PR.

As far as other issues to help with, we try and keep labels for good starter changes here: https://github.com/dolthub/dolt/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22

**Pull request merged**